spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/AbstractMessageChannelBinder.java

```diff
@@ -98,7 +103,7 @@ public AbstractMessageChannelBinder(boolean supportsHeadersNatively, String[] he
 			final P producerProperties) throws BinderException {
 		Assert.isInstanceOf(SubscribableChannel.class, outputChannel,
 				"Binding is supported only for SubscribableChannel instances");
-		PD producerDestination = createProducerDestinationIfNecessary(destination, producerProperties);
+		final ProducerDestination producerDestination = this.provisioningProvider.provisionProducerDestination(destination, producerProperties);
```

Comment:
Yeah. `this.provisioningProvider.provisionConsumerDestination` is really included in the `try...catch`.
See `doBindConsumer()`